core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -615,6 +615,42 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
     assertEquals(brokerStrs.mkString(","), nodeStrs.mkString(","))
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = Array("kraft"))
+  def testListNodesNotIncludingFencedBrokers(quorum: String): Unit = {
+    client = createAdminClient
+    val fencedBrokerId = brokers.last.config.brokerId
+    killBroker(fencedBrokerId)
+    Thread.sleep(10000) //sleep is needed to ensure the broker is fenced after the shutdown
```

Comment:
This is my concern as well. It seems to take at least a few seconds for the broker to become fenced after killing it and was hoping that 10000ms would be more than enough (need to rerun it a number of times in Jenkins to confirm). I understand there is a risk of it not being enough resulting in flakiness. I see that some of the integration tests also use TestUtils.retry() to retry when there is an assertion error, until the max timeout is reached. This still does not guarantee, but I wonder if that's a better solution with a larger max timeout, than the sleep.

Reply:
This was my concern too but the broker takes a few seconds to get fenced after being killed and 10000ms seems to be sufficient, although I had planned to run this test several times in Jenkins to confirm it and adjust if needed. 

I pushed another commit to merge these two test cases together, so that we only wait for a broker to get fenced once instead of twice. Also used TestUtils.retry to retry the request until we receive 2 of 3 brokers in the response or reach the max wait time. I set the max wait time to a larger value, in case it takes longer sometimes, but it shouldn't take more than 10s normally.